### DISCUSSION TITLE
Instrumentation in Uno

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9825] [Q&A]
[_Question from Discord_:](https://discord.com/channels/732297728826277939/734034054206193784/1019229914282606762)

So I've looked around for this already but sorry if this isn't answered anywhere, but I'm looking to add instrumentation to our WPF app that will be ported into either Avalonia or UNO (not sure which) but would like to know how the UNO-Platform has done it in any of theirs. 

The main use case will be just to measure performance of methods but don't want to litter the code manually with stopwatches and would ideally like to integrate it some third-party displayer, such as the AWS one, Azure Insights, New Relic e.t.c. 

Does anyone know any that can be integrated with UNO?

**Comment by jeromelaban** [comment]
Developing on Uno Platform is no different from developing for WPF, where most telemetry and analytics libraries should work as-is on Desktop platforms. The story may become different if you want to target non-desktop platforms like WebAssembly or mobile. Is that the case?

**Reply by oscarjaergren** [reply]
It ideally would be all of them in one go, but I would be happy for it to work with just the Windows version and explore a different or semi-integrated solution for the web and mobile portion. 

But a one-in all solution would be the best option
